odoo/http.py

```diff
@@ -1447,8 +1448,36 @@ def _dispatch_nodb():
                     return request._handle_exception(e)
                 return result
 
-            with request:
-                db = request.session.db
+            db = request.session.db
+
+            context_managers = [request]
+
+            if request.session.profile_session and db:
+                if 'set_profiling' in request.httprequest.path:
+                    _logger.info("Profiling disabled on route set_profiling")
+                elif request.session.profile_expiration < str(datetime.now()):
+                    # avoid having session profiling for too long if user forgets to disable profiling
+                    request.session.profile_session = None
+                    _logger.warning("Profiling expiration reached, disabling profiling")
+                elif request.httprequest.path.startswith('/longpolling') and odoo.multi_process:
+                    # disable profiling for gevent server
+                    _logger.info('Profiling disabled for longpolling in worker mode')
```

Comment:
Could be interresting if you want to profile long_polling or im_status. Only in worker mode because gevent breaks a little the thread logic used to retrieve stack. 

sys._current_frames()[threading.current_thread().ident] will return a keyError in a gevent "thread"

Reply:
But I actually think that it would be a good idea to avoid profiling longpolling most of the time.

Reply:
> Could be interresting if you want to profile longpolling or im_status. Only in worker mode because gevent breaks a little the thread logic used to retrieve stack.

Right, but I don't think that's useful: deploying longpolling or im_status in threaded mode is bonkers, and the behaviour will not match what you'd get in production.

> sys._current_frames()[threading.current_thread().ident] will return a keyError in a gevent "thread"

Maybe use gevent's threading thingie instead? I think it has one. Though I don't know if it correctly bridges `sys._current_frames`.

But I think just not observing longpolling would be better, at least in the first version.